cmd/generate/config/config.go

```diff
@@ -0,0 +1,271 @@
+				return errors.New("you must provide exactly one argument: the path to the repository")
+			}
+
+			path := args[0]
+
+			// Validate that the path is a real path on disk and accessible by the user
+			absPath, err := filepath.Abs(path)
+			if err != nil {
+				return err
+			}
+
+			if _, err := os.Stat(absPath); os.IsNotExist(err) {
+				return fmt.Errorf("the provided path does not exist: %w", err)
+			}
+
+			opts.path = absPath
+			return nil
+		},
+
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			// TODO: error checking based on given command
+
+			opts.outputPath, _ = cmd.Flags().GetString("output-path")
+			opts.isInteractive, _ = cmd.Flags().GetBool("interactive")
+
+			return run(opts)
+		},
+	}
+
+	cmd.PersistentFlags().StringP("output-path", "o", "./", "Directory to create the `.sauced.yaml` file.")
+	cmd.PersistentFlags().BoolP("interactive", "i", false, "Whether to be interactive")
+	return cmd
+}
+
+func run(opts *Options) error {
+	attributionMap := make(map[string][]string)
+
+	// Open repo
+	repo, err := git.PlainOpen(opts.path)
+	if err != nil {
+		return fmt.Errorf("Error opening repo: %w", err)
+	}
+
+	commitIter, err := repo.CommitObjects()
+
+	if err != nil {
+		return fmt.Errorf("Error opening repo commits: %w", err)
+	}
+
+	var uniqueEmails []string
+	err = commitIter.ForEach(func(c *object.Commit) error {
+		name := c.Author.Name
+		email := c.Author.Email
+
+		if !opts.isInteractive {
+			doesEmailExist := slices.Contains(attributionMap[name], email)
+			if !doesEmailExist {
+				// AUTOMATIC: set every name and associated emails
+				attributionMap[name] = append(attributionMap[name], email)
+			}
+		} else if !slices.Contains(uniqueEmails, email) {
+			uniqueEmails = append(uniqueEmails, email)
+		}
+		return nil
+	})
+
+	if err != nil {
+		return fmt.Errorf("Error iterating over repo commits: %w", err)
+	}
+
+	// INTERACTIVE: per unique email, set a name (existing or new or ignore)
+	if opts.isInteractive {
+		program := tea.NewProgram(initialModel(opts, uniqueEmails))
+		if _, err := program.Run(); err != nil {
+			return fmt.Errorf("Error running interactive mode: %w", err)
+		}
+	} else {
+		// generate an output file
+		// default: `./.sauced.yaml`
+		// fallback for home directories
+		if opts.outputPath == "~/" {
+			homeDir, _ := os.UserHomeDir()
+			err := generateOutputFile(filepath.Join(homeDir, ".sauced.yaml"), attributionMap)
+			if err != nil {
+				return fmt.Errorf("Error generating output file: %w", err)
+			}
+		} else {
+			err := generateOutputFile(filepath.Join(opts.outputPath, ".sauced.yaml"), attributionMap)
+			if err != nil {
+				return fmt.Errorf("Error generating output file: %w", err)
+			}
+		}
+	}
+
+	return nil
+}
+
+// Bubbletea for Interactive Mode
+
+type model struct {
+	textInput textinput.Model
+	help      help.Model
+	keymap    keymap
+
+	opts           *Options
+	attributionMap map[string][]string
+	uniqueEmails   []string
+	currentIndex   int
+}
```

Comment:
I'd move this (alongside the `type keymap struct{}` out to it's own file in the `config` package just to keep things organized. All this can stay as is, just drop it in a new file under `cmd/generage/config` and have

```go
package config
```

at the top!